core/src/main/java/tc/oc/pgm/filters/matcher/player/PlayerStateFilter.java

```diff
@@ -0,0 +1,31 @@
+package tc.oc.pgm.filters.matcher.player;
+
+import com.google.common.collect.ImmutableList;
+import java.util.Collection;
+import org.bukkit.event.Event;
+import tc.oc.pgm.api.filter.query.PlayerQuery;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.api.player.event.MatchPlayerDeathEvent;
+import tc.oc.pgm.spawns.events.PlayerSpawnEvent;
+
+public class PlayerStateFilter extends ParticipantFilter {
+
+  public static final PlayerStateFilter ALIVE = new PlayerStateFilter(true);
+  public static final PlayerStateFilter DEAD = new PlayerStateFilter(false);
+
+  private final boolean alive;
+
+  public PlayerStateFilter(boolean alive) {
+    this.alive = alive;
+  }
+
+  @Override
+  public Collection<Class<? extends Event>> getRelevantEvents() {
+    return ImmutableList.of(PlayerSpawnEvent.class, MatchPlayerDeathEvent.class);
```

Comment:
I'd use `ParticipantSpawnEvent` and `ParticipantDespawnEvent`, they won't work for player going from dead to obs tho, which is why i'd consider just making it `<interacting/>` instead which should be fully covered by those 2 events